cmd/otelcol/main.go

```diff
@@ -169,44 +155,79 @@ func checkRuntimeParams() {
 	// Check if memory ballast flag was passed
 	// If so, ensure memory ballast env var is not set
 	// Then set memory ballast and limit properly
-	ballastSize := getKeyValue(args, "--mem-ballast-size-mib")
+	ballastSize := getKeyValue(os.Args[1:], "--mem-ballast-size-mib")
 	if ballastSize != "" {
-		config = os.Getenv(ballastEnvVarName)
-		if config != "" {
-			log.Fatalf("Both %v and '--config' were specified, but only one is allowed", ballastEnvVarName)
+		if os.Getenv(ballastEnvVarName) != "" {
+			log.Fatalf("Both %v and '--mem-ballast-size-mib' were specified, but only one is allowed", ballastEnvVarName)
 		}
 		os.Setenv(ballastEnvVarName, ballastSize)
 	}
 	setMemoryBallast(memTotalSizeMiB)
 	setMemoryLimit(memTotalSizeMiB)
 }
 
-// Validate and set the configuration
-func setConfig() {
-	// Check if the config is specified via the env var.
-	config := os.Getenv(configEnvVarName)
-	// If not attempt to use a default config; supports Docker and local
-	if config == "" {
-		_, err := os.Stat(defaultDockerSAPMConfig)
-		if err == nil {
-			config = defaultDockerSAPMConfig
+// Set config file path environment variable to specified config file path flag.
+// Set the config file path flag to the specified config file path environment variable if not specified.
+// Set the config file path flag, config file path environment variable to default config file path if the config
+// file path flag, config file path environment variable and config YAML environment variable are not specified.
+func setConfigPath() {
+	configPathFlag := getKeyValue(os.Args[1:], "--config")
```

Comment:
What is `--config` is not first? We do not mandate an order with upstream collector do we?

Reply:
I don't follow your question, "What is --config is not first?".

Reply:
Sorry -- what _if_: `otelcol --metrics-addr 0.0.0.0:8123 --config foo.yaml`